### PR TITLE
Adding support for binary8

### PR DESCRIPTION
**Pull request by mfasi**
I'm creating a pull request so that we can comment on changes before rebasing.

### CODE REVIEW

File: src/cpfloat_definitions.h [hunk 1]
```diff
@@ -202,6 +202,19 @@ typedef struct {
    * exponent is larger than the maximum allowed by the storage format.
    */
   cpfloat_exponent_t emax;
+  /**
+   * @brief Minimum exponent of target format.
+   *
+   * @details The minimum values allowed are -126 and -1022 if the storage format
+   * is `float` or `double`, respectively. Smaller values are increase to the
```
Comment (mfasi) [inline src/cpfloat_definitions.h:209]: `increased`? Perhaps I would change this sentence to read `If a smaller value is chosen, this is changed to the minimum allowed value without warning.` And probably, the `emax` docstring should be changed accordingly – I find it's not the clearest description of what's happening.

File: src/cpfloat_template.h [hunk 2]
```diff
@@ -106,6 +106,7 @@ optstruct *init_optstruct() {
   fpopts->bitseed = NULL;
   fpopts->randseedf = NULL;
   fpopts->randseed = NULL;
+  fpopts->emin = -99999;
```
Comment (mfasi) [inline src/cpfloat_template.h:109]: Not sure I like this magic constant – would 0 do as well here? `emin` should be strictly negative anyway.

File: src/cpfloat_template.h [hunk 3]
```diff
@@ -279,6 +280,10 @@ static inline int VALIDATE_INPUT(const optstruct *fpopts) {
   if (fpopts->flip != CPFLOAT_NO_SOFTERR && (fpopts->p > 1 || fpopts->p < 0))
     return 5;
 
+  /* Return -6 if emin is invalid (either nonnegative or too small). */
+  if (fpopts->emin < DEFEMIN || fpopts->emin >= 0)
```
Comment (mfasi) [inline src/cpfloat_template.h:284]: No changes needed here, as far as I can tell.

File: src/cpfloat_template.h [hunk 4]
```diff
@@ -304,7 +309,14 @@ static inline FPPARAMS COMPUTE_GLOBAL_PARAMS(const optstruct *fpopts,
   }
 
   /* Derived floating point parameters. */
-  int emin = 1-emax;
+  int emin = fpopts->emin;
+  /* If emin is not set by user, set it to the default 1-emax. */
+  if (emin == -99999)
```
Comment (mfasi) [inline src/cpfloat_template.h:314]: This `-9999` would become a `0`, I think.

File: src/cpfloat_template.h [hunk 5]
```diff
@@ -304,7 +309,14 @@ static inline FPPARAMS COMPUTE_GLOBAL_PARAMS(const optstruct *fpopts,
   }
 
   /* Derived floating point parameters. */
-  int emin = 1-emax;
+  int emin = fpopts->emin;
+  /* If emin is not set by user, set it to the default 1-emax. */
```
Comment (mfasi) [inline src/cpfloat_template.h:313]: Should this be `the user`?

File: src/cpfloat_template.h [hunk 6]
```diff
@@ -1199,7 +1211,7 @@ GENERATE_UNIVARIATE_MATH_H(lgamma, lgamma)
    if (temp == FP_ILOGB0 || temp == FP_ILOGBNAN || temp == INT_MAX) {          \
      params.precision = DEFPREC-1;                                             \
      params.emax = DEFEMAX;                                                    \
-     params.emax = DEFEMIN;                                                    \
+     params.emin = DEFEMIN;                                                    \
```
Comment (mfasi) [inline src/cpfloat_template.h:1214]: Thanks for this – can you change the message of the last commit to

`Fixed the emin bug (closes #2)`

**Review by mfasi — COMMENTED** [review]
Just a few minor comments. The main issue I see is that the tests are currently failing, but that seems an unrelated problem – I'll look into it.

**Comment by mfasi** [comment]
> The main issue I see is that the tests are currently failing, but that seems an unrelated problem – I'll look into it.

I think I've found the issue and fixed it locally. I can push this to `main` if you're happy to rebase `dev` onto it, or I can wait for you to be done with this PR and rebase my change onto `dev`.

**Comment by mmikaitis** [comment]
> > The main issue I see is that the tests are currently failing, but that seems an unrelated problem – I'll look into it.
> 
> I think I've found the issue and fixed it locally. I can push this to `main` if you're happy to rebase `dev` onto it, or I can wait for you to be done with this PR and rebase my change onto `dev`.

Please push it to main. I will rebase onto it once done here.

**Comment by mfasi** [comment]
Looks good, thanks! I'll rebase and merge – the code should pass the revised tests, and if not, we can fix it later.

**Comment by mfasi** [comment]
Mhh – it didn't quite work. It seems that rebase and merge doesn't actually rebase on `main`... Will do it by hand.

**Pull request merged**